lib/data_cleanup/rules/section/fix_duplicate_number.rb

```diff
@@ -10,18 +10,74 @@ def description
         end
 
         def call
-          data = ::Section.group(:number, :phase_id)
-                          .count
-                          .select { |k,v| v > 1 }
-          data.each do |values, count|
-            number, phase_id = *values
-            ids = ::Section.where(phase_id: phase_id)
-                           .order("number ASC, created_at ASC")
-                           .pluck(:id)
-            phase = ::Phase.find(phase_id)
+          # A set of unique phase_ids that contain Sections with duplicate numbers
```

Comment:
@xsrust this is the biggie. Could you please go through it and make sure it fits your expectations?